app/models/spree/gateway/cardknox.rb

```diff
@@ -0,0 +1,18 @@
+module Spree
+    class Gateway::Cardknox < Gateway
+      preference :api_key, :string
+  
+      def provider_class
+        ActiveMerchant::Billing::CardknoxGateway
+      end
+    end
+end
+
+# A hack because ActiveMerchant's CardknoxGateway class returned an error when a paymentmethod was submitted that didn't respond to track_data and manual_entry methods
+# Can be removed when Pull Request #2580 is merged to ActiveMerchant
+
+module AdditionalMethods
+    attr_accessor :track_data, :manual_entry
```

Comment:
I don't want to add anything to solidus_gateway which monkey patches solidus_core. Maybe this gateway should instead be in its own extension.

Reply:
Valid point, John. I think we should start to encourage people to write separate extensions for payment providers in stead of bloating this (already arguable) gem 👌🏻

Reply:
Hmmm I was going to remove the monkey patch and just overwrite the offending methods myself. But if you'd prefer not to add more gateways I can look into creating a new extension...

Reply:
I think an extension on its own would be better.

You could then add you own payment source class that inherits from `Spree::PaymentSource` with your extra methods added.

Also you don't have to deal with legacy Solidus versions and old Active Merchant versions.